### PR TITLE
Fix Flyout placement for versions <RS5 

### PR DESCRIPTION
**Pull request by khetanashita**
Some options for flyout placement were added in RS5 (like FlyoutPlacementMode::BottomEdgeAlignedLeft, TopEdgeAlignedLeft and LeftEdgeAlignedTop). 

Adding a check to avoid failure for <RS5 devices.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/2676)

### CODE REVIEW

**Comment by chrisglein** [comment]
What's going on with the commit history of this PR? (264 commits)

**Comment by ahimberg** [comment]
> What's going on with the commit history of this PR? (264 commits)

This seems to happen with people that submit a pull request from their fork/master instead of making a branch in their repo.  But yeah because of 264 commits you can't use codeflow to review the changes

**Comment by chrisglein** [comment]
> What's going on with the commit history of this PR? (264 commits)
> 
> This seems to happen with people that submit a pull request from their fork/master instead of making a branch in their repo.  But yeah because of 264 commits you can't use codeflow to review the changes

Ah, that explains it. Good candidate for something to add to a contribution guide (it's mentioned in react-native's [here](https://github.com/facebook/react-native/blob/master/CONTRIBUTING.md#contributing-code), CoreCLR's [here](https://github.com/dotnet/coreclr/blob/master/Documentation/project-docs/contributing-workflow.md), and WinUI's [here](https://github.com/microsoft/microsoft-ui-xaml/blob/master/docs/contribution_workflow.md)

**Comment by khetanashita** [comment]
> > What's going on with the commit history of this PR? (264 commits)
> > This seems to happen with people that submit a pull request from their fork/master instead of making a branch in their repo.  But yeah because of 264 commits you can't use codeflow to review the changes
> 
> Ah, that explains it. Good candidate for something to add to a contribution guide (it's mentioned in react-native's [here](https://github.com/facebook/react-native/blob/master/CONTRIBUTING.md#contributing-code), CoreCLR's [here](https://github.com/dotnet/coreclr/blob/master/Documentation/project-docs/contributing-workflow.md), and WinUI's [here](https://github.com/microsoft/microsoft-ui-xaml/blob/master/docs/contribution_workflow.md)

My bad. I'll make sure to work off of a dedicated branch rather than master.

File: vnext/ReactUWP/Views/FlyoutViewManager.cpp [hunk 1]
```diff
@@ -23,61 +23,88 @@ struct json_type_traits<winrt::FlyoutPlacementMode>
...
+      {
+        placement = winrt::FlyoutPlacementMode::Left;
+      }
+      else if (json == "full")
+      {
+        placement = winrt::FlyoutPlacementMode::Full;
+      }
+      else
+      {
+        placement = winrt::FlyoutPlacementMode::Right;
```
[the remark this answers is not in the document]
Comment (khetanashita) [reply vnext/ReactUWP/Views/FlyoutViewManager.cpp]: With the new changes, always defaulting to right.

File: vnext/ReactUWP/Views/FlyoutViewManager.cpp [hunk 2]
```diff
@@ -23,61 +23,88 @@ struct json_type_traits<winrt::FlyoutPlacementMode>
...
-    else if (json == "right-edge-aligned-bottom")
-    {
-      placement = winrt::FlyoutPlacementMode::RightEdgeAlignedBottom;
-    }
-    else if (json == "full")
+
+    if (auto flyoutBase5 = winrt::Flyout().try_as<winrt::IFlyoutBase5>())
     {
-      placement = winrt::FlyoutPlacementMode::Full;
+      if (json == "top")
```
[the remark this answers is not in the document]
Comment (khetanashita) [reply vnext/ReactUWP/Views/FlyoutViewManager.cpp]: Switched to using a map. Looks way better now.

File: vnext/ReactUWP/Views/FlyoutViewManager.cpp [hunk 3]
```diff
@@ -17,70 +17,44 @@ namespace winrt {
   using namespace Windows::UI::Xaml::Interop;
 }
 
+static const std::map<std::string, winrt::FlyoutPlacementMode> placementModeMinVersion = {
```
Comment (ahimberg) [inline vnext/ReactUWP/Views/FlyoutViewManager.cpp]: may want to use unordered_map as slightly cheaper.  don't think either can be constexpr unfortunately

**Pull request merged**